projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+func main() {
+	c := http.Client{Timeout: time.Duration(1) * time.Second}
+
+	url := "http://localhost:8080"
+	max_tries := 5
+
+	fmt.Println("----------------------------")
+
+	for i := 0; i < max_tries; i++ {
+		resp, err := c.Get(url)
+		if err != nil {
+			fmt.Println("Error from request", err)
+			return
+		}
+
+		if resp.StatusCode == http.StatusTooManyRequests {
+			retryAfter := resp.Header.Get("Retry-After")
+			retryAfterInteger, err := strconv.Atoi(retryAfter)
+			if err != nil {
+				// This is where we give up if we do not have the amount of time as an integer. We would not like to wait for it as it is uncertain how long is it gonna be.
+				fmt.Println("Error parsing string")
+				return
+			}
```

Comment:
There are a few different things the server may return in this header - looking at the server code, what things could it return? When the value isn't an integer, what do they mean, and is there something reasonable we can do about them?